include/battle.h

```diff
@@ -221,7 +221,8 @@ struct SpecialStatus
     u8 faintedHasReplacement:1;
     u8 focusBanded:1;
     u8 focusSashed:1;
-    u8 unused:2;
+    bool8 ejectButtonSwitch:1;
+    bool8 ejectPackSwitch:1;
```

Comment:
Those are ai specific right? If yet then it is probably better to have them in the AI struct.

Reply:
That definitely might make more sense! I wanna check though, based on issues like #5229, #5227, and #5219, I think it's a very real target to try to make functions like `ShouldSwitch` and `GetBestMonIntegrated` easy to adapt to the player's perspective where we can to minimize the workload when we eventually try to make those usable by the AI for the player's perspective.

Would having these in the AI struct disrupt that concept? I wasn't sure which is why I put it here, but if you think that wouldn't cause issues I will happily move it :)

Reply:
I don't see any issues with it. It is also ai specific so I feel it makes more sense there.